addons/website/static/src/interactions/header/base_header.js

```diff
@@ -0,0 +1,206 @@
+            for (const collapseEl of collapseEls) {
+                Collapse.getOrCreateInstance(collapseEl).hide();
+            }
+        }
+    }
+
+    //--------------------------------------------------------------
+    // Animation Handlers
+    //--------------------------------------------------------------
+
+    adaptToHeaderChange() {
+        for (const callback of this.services.menu_callback.updateCallbacks) {
+            callback();
+        }
+        this.adjustMainPadding();
+    }
+
+    adaptToHeaderChangeLoop(addCount = 0) {
+        if (this.isDestroyed) {
+            return;
+        }
+        this.adaptToHeaderChange();
+        this.transitionCount = Math.max(0, this.transitionCount + addCount);
+
+        // As long as we detected a transition start without its related
+        // transition end, keep updating the main padding top.
+        if (this.transitionCount > 0) {
+            this.el.classList.add("o_transitioning");
+            window.requestAnimationFrame(() => this.adaptToHeaderChangeLoop());
+
+            // The normal case would be to have the transitionend event to be
+            // fired but we cannot rely on it, so we use a timeout as fallback.
+            if (addCount !== 0) {
+                clearTimeout(this.changeLoopTimer);
+                this.changeLoopTimer = setTimeout(() => this.adaptToHeaderChangeLoop(- this.transitionCount), 500);
+            }
+        } else {
+            // When we detected all transitionend events, we need to stop the
+            // setTimeout fallback.
+            this.el.classList.remove("o_transitioning");
+            clearTimeout(this.changeLoopTimer);
+        }
+    }
+
+    //--------------------------------------------------------------
+    // Animation Trigger
+    //--------------------------------------------------------------
+
+    transformShow() {
+        this.isVisible = true;
+        this.el.style.transform = this.atTop ? "" : `translate(0, -${this.topGap}px)`;
+        this.adaptToHeaderChangeLoop(1);
+    }
+
+    transformHide() {
+        this.isVisible = false;
+        this.el.style.transform = "translate(0, -100%)";
+        this.adaptToHeaderChangeLoop(1);
+    }
+
+    //--------------------------------------------------------------
+    // Change Handlers
+    //--------------------------------------------------------------
+
+    adjustPosition() {
+        // When the url contains #aRandomSection, prevent the navbar to overlap
+        // on the section, for this, we scroll as many px as the navbar height.
+        document.scrollingElement.scrollBy(0, - this.el.offsetHeight);
+    }
+
+    adjustScrollbar() {
+        compensateScrollbar(this.el, this.cssAffixed, false, 'right');
+    }
+
+    adjustMainPadding() {
+        if (this.isOverlay) {
+            return;
+        }
+        this.mainEl.style.setProperty("padding-top", this.cssAffixed ? this.getHeaderHeight() + "px" : "");
```

Comment:
and I think this is were the cache should be useful. (There was a ` ||= ` in the old code.)